### PR TITLE
Guard against reinjection of items when pushing items into the grid from CPUCluster

### PR DESCRIPTION
**Pull request by shartte**
Potential fix for #5158: Guard against re-insertion of items into the crafting CPU when dumping items into the network after a job was completed.

### CODE REVIEW

File: src/main/java/appeng/me/cluster/implementations/CraftingCPUCluster.java [hunk 1]
```diff
@@ -397,6 +400,23 @@ private void completeJob() {
...
+                errorMessage.append(" - ").append(stack).append("\n");
+            }
+            errorMessage.append("Current inventory:\n");
+            for (IAEItemStack stack : inventory.getItemList()) {
+                errorMessage.append(" - ").append(stack).append("\n");
+            }
+            errorMessage.append("Final output: ").append(finalOutput.copy().setStackSize(this.startItemCount));
+            throw new RuntimeException(errorMessage.toString());
+        }
+        this.waitingFor.resetStatus();
```
Comment (yueh) [inline src/main/java/appeng/me/cluster/implementations/CraftingCPUCluster.java:418]: Wouldn't that be actually dead code/useless?

Comment (shartte) [reply src/main/java/appeng/me/cluster/implementations/CraftingCPUCluster.java:418]: It's for "the future" when we actually remove the fail-fast code. I do want to clear that list :-P

Comment (yueh) [reply src/main/java/appeng/me/cluster/implementations/CraftingCPUCluster.java:418]: It's still wrong. A crafting job can never be complete, when it's still waiting on more items. So it's always reseting an empty list, which is a bit pointless. Or it's actually an unfixed bug.

Comment (shartte) [reply src/main/java/appeng/me/cluster/implementations/CraftingCPUCluster.java:418]: It still shows what the invariant is. We _can_ also replace this with Precondtion.checkState(waitingFor.isEmpty()) if you prefer that. I just don't  want this to be forgotten when we do remove the in-depth check .

**Pull request merged**